support_branding/models/res_company.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2023 Sunflower IT
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, models
+from odoo.exceptions import AccessError
+
+
+class ResCompany(models.Model):
+    _inherit = "res.company"
+
+    @api.model
+    def get_support_branding_config_param_data(self):
+        if not self.env.user.has_group("base.group_user"):
+            raise AccessError(
+                _(
+                    "You are not allowed to access this "
+                    "functionality, please contact Admin for "
+                    "more support"
+                )
+            )
+        self.env.cr.execute(
+            "select key, value from ir_config_parameter where key ilike "
+            "'%support_%';"
+        )
```

Comment:
why don't you just do a `sudo().search()` here? I think it would also be good to search for `['|', ('company_id', '=', False), ('company_id', 'in', company_ids)]` with order `company_id nulls last` to support different branding per company

Reply:
Not sure why we would use **company_id** because its not included in the ```ir_config_parameter``` model. @thomaspaulb @hbrunn unless I'm missing a bigger picture here with the requested changes.

Reply:
From what I understand, currently the code uses the same config parameters for each company, like it used to do anyway. Making the module with multi-company support would not be part of the fix I guess, but it would be cool.

If you would do that, maybe what could be done is use queries like this:
```
self.env.cr.execute(
    "select key, value from ir_config_parameter where key ilike "
    "'%support_%i_%';" % company_id
)
```
But then I suppose you need to move the view away from the general settings and maybe into the company form, then you should have a company id. Or maybe leave it and use `self.env.context['allowed_company_ids']` somehow?

Reply:
From what I understand, currently the code uses the same config parameters for each company, like it used to do anyway. Making the module with multi-company support would not be part of the fix I guess, but it would be cool.

If you would do that, maybe what could be done is use queries like this:
```
self.env.cr.execute(
    "select key, value from ir_config_parameter where key ilike "
    "'support_%i_%';" % company_id
)
```
But then I suppose you need to move the view away from the general settings and maybe into the company form, then you should have a company id. Or maybe leave it and use an id from `self.env.context['allowed_company_ids']` somehow.

Reply:
In general it seems to me a too obscure feature, since this module is targeted at explaining users where to direct support queries, and for 90% of cases, even in a multi-company instance, that whole instance is supported by just one Odoo partner.

Reply:
fine with me